src/components/ShareTrajectoryModal/index.tsx

```diff
@@ -74,37 +68,79 @@ const ShareTrajectoryModal = ({
             inputAsNumber + displayTimes.roundedTimeStep >=
             displayTimes.roundedLastFrameTime
         ) {
+            // set time to last frame time
             timeValue =
                 displayTimes.roundedLastFrameTime -
                 displayTimes.roundedTimeStep;
         } else {
-            timeValue = roundToTimestepPrecision(
+            timeValue = roundToTimeStepPrecision(
                 inputAsNumber,
                 displayTimes.roundedTimeStep
             );
         }
-        setLastEnteredNumber(timeValue);
-        setUrl(
-            editUrlParams(
-                window.location.href,
-                timeValue.toString(),
-                URL_PARAM_KEY_TIME
-            )
-        );
+        setUserEnteredTime(timeValue);
     };
 
-    const copyToClipboard = async (): Promise<void> => {
-        try {
-            await navigator.clipboard.writeText(url);
-        } catch (err) {
-            console.error("Failed to copy text: ", err);
-        }
-    };
+    const UrlLinkPanel: JSX.Element = (
```

Comment:
is there anything reusable about this UrlLinkPanel that it could go into its own tsx file?